src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -332,19 +315,35 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                     break;
             }
 
+            if (isIntegratedSecurity)
+            {
+                string hostName = details.ServerName;
```

Comment:
The details.* can be directly dereferenced while calling the function. That can avoid the allocations for these three variables.